File: RoRClient/Model/Models/Square.cs
﻿using RoRClient.ViewModel.Helper;
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace RoRClient.Model.Models
{
    public class Square : ModelBase
    {
        /// <summary>
        /// Klasse für ein Feld, dass auf der Map liegt und eine Position hat.
        /// Auf dem Feld platzierte Objekte können auf die jeweilige Position zugreifen
        /// </summary>

        public Square() : base()
        {

        }

        private int posX;
        public int PosX
        {
            get
            {
                return posX;
            }
            set
            {
                if(posX != value)
                {
                    int temp = posX;
                    posX = value;
                    NotifyPropertyChanged("XPos", temp, posX);
                }
            }
        }

        private int posY;

        public int PosY
        {
            get
            {
                return posY;
            }
            set
            {
                if(posY != value)
                {
                    int temp = posY;
                    posY = value;
                    NotifyPropertyChanged("YPos", temp, posY);
                }
            }
        }

        public Square (int xPos, int yPos)
        {
            this.posX = xPos;
            this.posY = yPos;
        }

        IPlaceableOnSquare placeableOnSquare = null;
        public IPlaceableOnSquare PlaceableOnSquare
        {
            get
            {
                return placeableOnSquare;
            }
            set
            {
                if(placeableOnSquare != value)
                {
                    IPlaceableOnSquare temp = placeableOnSquare;
                    placeableOnSquare = value;
                    NotifyPropertyChanged("PlaceableOnSquare", temp, placeableOnSquare);
                }
            }
        }
    }
}

File: RoRServer/src/main/java/models/Square.java
package models;

/**
 * Klasse für ein Feld (Square), dass auf der Map liegt und eine Position hat.
 * Auf dem Feld platzierte Objekte können auf die jeweilige Position zugreifen.
 */
public class Square {

	private PlaceableOnSquare placeableOnSquare = null;
	private int xIndex;
	private int yIndex;
	
	public Square(int xIndex, int yIndex) {
		this.xIndex = xIndex;
		this.yIndex = yIndex;
	}
	
	public void setPlaceable(PlaceableOnSquare placeable) {
		//if (placeableOnSquare == null ) {
			this.placeableOnSquare = placeable;
		//}
	}
	
	public PlaceableOnSquare getPlaceableOnSquare(){
		return placeableOnSquare;
	}
	
	public int getXIndex(){
		return xIndex;
	}
	
	public int getYIndex(){
		return yIndex;
	}
}
